lib/elixir/src/elixir_exp.erl

```diff
@@ -230,8 +230,16 @@ expand({fn, Meta, Pairs}, E) ->
 
 expand({'cond', Meta, [KV]}, E) ->
   assert_no_match_or_guard_scope(Meta, 'cond', E),
-  {EClauses, EC} = elixir_exp_clauses:'cond'(Meta, KV, E),
-  {{'cond', Meta, [EClauses]}, EC};
+  {EBody, EC} = elixir_exp_clauses:'cond'(Meta, KV, E),
+  [{do, EClauses}] = EBody,
+  case lists:last(EClauses) of
```

Comment:
@josevalim that makes sense. I moved to a separate function and also fixed a tiny thing where I was using the `Meta` of the whole `cond` in the error, instead of the `Meta` of the `'->'` clause.